### PR TITLE
Add Donation Entity

### PR DESCRIPTION
**Pull request by gabrielpoca**
In this change, we add donations to the subgraph. For donations, we need to include the `data` field from the claim params in the events. Unfortunately, the test code for the mappings has a lot of duplication because it's AssemblyScript and not Typescript. I tried many ways to remove that duplication into functions but I couldn't find a way. The compiler doesn't allow me and I've already spent too many hours on it.

Extra: Fixed docker deployment.

### CODE REVIEW

File: subgraph/src/mappings/vault.ts [hunk 1]
```diff
@@ -1,16 +1,88 @@
...
+      .times(event.params.burnedShares)
+      .div(event.params.amount);
+
+    totalClaimedShares = totalClaimedShares.plus(claimedShares);
+    deposit.shares = deposit.shares.minus(claimedShares);
+    deposit.save();
+
+    if (
+      event.params.to.equals(
+        ByteArray.fromHexString("0x4940c6e628da11ac0bdcf7f82be8579b4696fa33")
```
Comment (naps62) [inline subgraph/src/mappings/vault.ts:50]: what's this address?

File: subgraph/tests/integreation.test.ts [hunk 2]
```diff
@@ -0,0 +1,392 @@
+import { log, ethereum, Address, Bytes, BigInt } from "@graphprotocol/graph-ts";
```
Comment (naps62) [inline subgraph/tests/integreation.test.ts:1]: typo in the file name

File: subgraph/src/mappings/vault.ts (hunk 1 above)
Comment (gabrielpoca) [reply subgraph/src/mappings/vault.ts:50]: Nice catch. I forgot to do this. It needs to come from an event emitted by the vault.

File: subgraph/tests/integreation.test.ts (hunk 2 above)
Comment (gabrielpoca) [reply subgraph/tests/integreation.test.ts:1]: 👍

File: subgraph/src/mappings/vault.ts [hunk 3]
```diff
@@ -1,16 +1,88 @@
...
+  claimer.save();
+  vault.save();
+}
 
 export function handleDepositMinted(event: DepositMinted): void {
-  const foundationId = event.params.groupId.toHexString();
-  const depositId = event.params.id.toHexString();
-  const claimerId = event.params.claimerId.toHexString();
-  const vaultId = event.address.toHexString();
+  const foundationId = event.params.groupId.toString();
```
Comment (gabrielpoca) [inline subgraph/src/mappings/vault.ts:82]: @naps62 I just realized that this won't work with multiple vaults. The groupId is an incremented number that will exist in all vaults eventually. I need to use the vault id as well.

**Comment by gabrielpoca** [comment]
@naps62 updated

File: contracts/Vault.sol [hunk 4]
```diff
@@ -137,6 +137,13 @@ contract Vault is
     //
     // IVault
     //
+    function setTreasury(address _treasury)
+        external
+        override(IVault)
+        requiresTrust
```
Comment (naps62) [inline contracts/Vault.sol:143]: you'll need to change this trust part though. it's been refactored in favor of AccessControl

**Review by naps62 — APPROVED** [review]

**Pull request merged**